src/serac/physics/utilities/functional/domain_integral_shared.hpp

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include "serac/physics/utilities/functional/tensor.hpp"
+
+namespace serac {
+
+namespace domain_integral {
+
+/**
+ * @brief Computes the arguments to be passed into the q-function (shape function evaluations)
+ * By default:
+ *  H1 family elements will compute {value, gradient}
+ *  Hcurl family elements will compute {value, curl}
+ *  TODO: Hdiv family elements will compute {value, divergence}
+ *  TODO: L2 family elements will compute value
+ *
+ * In the future, the user will be able to override these defaults
+ * to omit unused components (e.g. specify that they only need the gradient)
+ *
+ * @param[in] u The DOF values for the element
+ * @param[in] xi The position of the quadrature point in reference space
+ * @param[in] J The Jacobian of the element transformation at the quadrature point
+ * @tparam element_type The type of the element (used to determine the family)
+ */
+template <typename element_type, typename T, int dim>
+SERAC_HOST_DEVICE auto Preprocess(T u, const tensor<double, dim> xi, const tensor<double, dim, dim> J)
```

Comment:
Why are `Preprocess` and `Postprocess` capitalized?

Reply:
In this PR, they're capitalized for consistency with the earlier implementation (which should have been lowercase)

Reply:
I assumed this was because this will ideally be upstreamed to MFEM, so we're following their convention.

Reply:
I think it's just an oversight on my initial implementation

Reply:
We should put all of the `Functional` stuff in an `mfem_extensions` namespace to make this clearer.